react-navigation/react-navigation
# fix: center align TabBarIndicator when width is specified

Please provide enough information so that others can review your pull request.

**Motivation**

When width is set for the TabBarIndicator, it does not align correctly.
<img width="317" alt="before" src="https://github.com/react-navigation/react-navigation/assets/59786076/f3168e1b-f24b-45e8-a712-dcdec12b287a">

By allowing the getTranslateX function to accept a width parameter, it will now adjust the calculation of the outputRange to ensure the TabBarIndicator is properly centered when width is provided as a number.


```
<TabBar
renderIndicator={(props) => <TabBarIndicator {...props} width={16} />}
....
/>
```


<img width="321" alt="after" src="https://github.com/react-navigation/react-navigation/assets/59786076/02fd22db-ef46-42c3-8637-0b5a78c0fae3">



**Test plan**

1. Apply a fixed width to the TabBarIndicator.
2. Verify that the TabBarIndicator is centered within its container.
